app/controllers/provider/admin/backend_apis_controller.rb

```diff
@@ -42,7 +42,13 @@ def update
   end
 
   def destroy
-    # TODO
+    if @backend_api.services.any?
+      flash[:error] = "Backend API could not be deleted"
+      render :edit
+    else
+      @backend_api.destroy
```

Comment:
What if the `destroy` returns `false`?

Reply:
```ruby
def destroy
  if @backend_api.services.empty? && @backend_api.destroy
    return redirect_to provider_admin_dashboard_path, notice: 'Backend API deleted'
  else
    flash[:error] = "Backend API could not be deleted"
    render :edit
  end
end

Reply:
Then it won't be deleted 😃
OK I'll change it